packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -65,31 +82,66 @@ export const withReactQuery = (Wrapped, options = {}) => {
          * @private
          */
         static async doInitAppState({res, appJSX}) {
-            const queryClient = (res.locals.__queryClient =
-                res.locals.__queryClient || new QueryClient(queryClientConfig))
+            // Create a separate query client just for discovery to avoid shared state
+            const discoveryQueryClient = new QueryClient(queryClientConfig)
+
+            // The actual query client that will be used for the real render
+            const queryClient = new QueryClient(queryClientConfig)
+            res.locals.__queryClient = queryClient
 
             res.__performanceTimer.mark(PERFORMANCE_MARKS.reactQueryPrerender, 'start')
-            // Use `ssrPrepass` to collect all uses of `useQuery`.
-            await ssrPrepass(appJSX)
-            res.__performanceTimer.mark(PERFORMANCE_MARKS.reactQueryPrerender, 'end')
-            const queryCache = queryClient.getQueryCache()
-            const queries = queryCache.getAll().filter((q) => q.options.enabled !== false)
-            await Promise.all(
-                queries.map((q, i) => {
-                    // always include the index to avoid duplicate entries
-                    const displayName = q.meta?.displayName ? `${q.meta?.displayName}:${i}` : `${i}`
-                    res.__performanceTimer.mark(
-                        `${PERFORMANCE_MARKS.reactQueryUseQuery}::${displayName}`,
-                        'start'
-                    )
-                    return q
-                        .fetch()
+
+            try {
+                // Set up a listener to capture queries as they're created
+                const listener = createQueryPrefetchListener(discoveryQueryClient)
+
+                // Create a completely separate component tree for discovery
+                // We're using a new instance of all providers to ensure isolation
+                // potential perfomance impact since we are deepcloning a trea here???
+                const discoveryApp = (
+                    <QueryClientProvider client={discoveryQueryClient}>
+                        {/* We use a deep clone by recreating the entire tree structure */}
+                        {React.cloneElement(appJSX, {key: 'discovery-phase'})}
+                    </QueryClientProvider>
```

Comment:
As we are calling renderToStaticMarkup, React will complain about hydration mismatch. This is because when we call renderToStaticMarkup for first render, React internally will mutate the jsx tree, later on when renderToString use the jsx, it is the mutated jsx. Meanwhile the client when it tried to hydrate, it will use the `non-mutated` version of the jsx tree. 
Create a complete tree here will prevent that scenario
![image](https://github.com/user-attachments/assets/b35d19c3-605d-4d0b-b569-a5df1ad12ed2)